src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -1464,24 +1464,60 @@ public override IOperation VisitConditional(IConditionalOperation operation, int
                 // result - capture
 
                 SpillEvalStack();
-                int captureId = captureIdForResult ?? _captureIdDispenser.GetNextId();
 
                 BasicBlockBuilder whenFalse = null;
                 VisitConditionalBranch(operation.Condition, ref whenFalse, sense: false);
 
-                VisitAndCapture(operation.WhenTrue, captureId);
-
                 var afterIf = new BasicBlockBuilder(BasicBlockKind.Block);
-                LinkBlocks(CurrentBasicBlock, afterIf);
-                _currentBasicBlock = null;
+                IOperation result;
+
+                // Specialy handle cases with "throw" as operation.WhenTrue or operation.WhenFalse. We don't need to create an additional
+                // capture for the result because there won't be any result from the throwing branches.
+                if (operation.WhenTrue is IConversionOperation whenTrueConversion && whenTrueConversion.Operand.Kind == OperationKind.Throw)
+                {
+                    IOperation rewrittenThrow = Visit(whenTrueConversion.Operand);
+                    Debug.Assert(rewrittenThrow.Kind == OperationKind.None);
```

Comment:
Do we know what node this originally corresponds to, and do we have a bug tracking supporting it natively?

Reply:
> Do we know what node this originally corresponds to, and do we have a bug tracking supporting it natively?

This is not a bug, but by design behavior. It is created by the CFG builder for throw expression. This was discussed at design meeting. I believe that after this change it will be exposed only for error scenarios, i.e. erroneous use of a throw expression.

---
In reply to: [193964473](https://github.com/dotnet/roslyn/pull/27612#discussion_r193964473) [](ancestors = 193964473)